thorlcr/activities/thdiskbaseslave.cpp

```diff
@@ -430,10 +430,13 @@ void CDiskWriteSlaveActivityBase::close()
             outraw.clear();
         }
 
+        Owned<IFileIO> tmpFileIO;
         {
             CriticalBlock block(statsCs);
             mergeStats(fileStats, outputIO);
-            outputIO.clear();
+
+            // ensure it is released/destroyed after releasing crit, since the IFileIO might involve a final copy and take considerable time.
+            tmpFileIO.setown(outputIO.getClear());
```

Comment:
This seems fine to get out from under the statsCs.  But does it mean the timing info in the stats may not be correct ?  Is there a way to wait for the final copy to end and then do the mergeStats() ?

Reply:
as it was / as it stands - it won't make any difference.
Because the way the statistics are collected here, it will be getting the stats for 1 IFileIO, which is the primary file written, and disregards the timing for the final copy.

So this change doesn't fix that, but you're right that it should
I'll open a separate JIRA to look at how fix that, it will be slightly more complicated than this fix.
@mckellyln

Reply:
ok, sounds good.

Reply:
For reference new JIRA: https://track.hpccsystems.com/browse/HPCC-23283